service/src/main/java/tds/session/services/impl/SessionServiceImpl.java

```diff
@@ -23,4 +23,11 @@ public SessionServiceImpl(SessionRepository sessionRepository) {
     public Optional<Session> getSessionById(UUID id) {
         return sessionRepository.getSessionById(id);
     }
+
+    @Override
+    public void pause(final UUID sessionId, final String newStatus) {
+        // TODO:  Throw an exception if attempting to pause a Session that is already paused?
```

Comment:
If we just ignore that fact and say if it is already paused then it stays paused.  From a consumer point of view, the test is paused after they call this so that is a good thing.  Not sure what they would do with an Exception because presumably if they catch it and do something it would either be to ignore it because the test is paused after all, or not handle it and the Exception bubbles up and what they were trying to do stops.

Possibly a legitimate use-case when this might occur is a race condition between a proctor pausing a session for a student and a student pausing it themselves.